File: _work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-python/src/iii_sandbox/env.py
from __future__ import annotations

from .client import HttpClient


class EnvManager:
    def __init__(self, client: HttpClient, sandbox_id: str):
        self._client = client
        self._sandbox_id = sandbox_id

    async def get(self, key: str) -> dict:
        return await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/env/get", {"key": key}
        )

    async def set(self, vars: dict[str, str]) -> dict:
        return await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/env", {"vars": vars}
        )

    async def list(self) -> dict:
        return await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/env"
        )

    async def delete(self, key: str) -> dict:
        return await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/env/delete", {"key": key}
        )

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-rust/src/env.rs
use std::collections::HashMap;
use std::sync::Arc;

use crate::client::HttpClient;
use crate::error::Result;
use crate::types::{EnvDeleteResponse, EnvGetResponse, EnvListResponse, EnvSetResponse};

pub struct EnvManager {
    client: Arc<HttpClient>,
    sandbox_id: String,
}

impl EnvManager {
    pub fn new(client: Arc<HttpClient>, sandbox_id: String) -> Self {
        Self { client, sandbox_id }
    }

    pub async fn get(&self, key: &str) -> Result<EnvGetResponse> {
        self.client
            .post(
                &format!("/sandbox/sandboxes/{}/env/get", self.sandbox_id),
                Some(&serde_json::json!({ "key": key })),
            )
            .await
    }

    pub async fn set(&self, vars: HashMap<String, String>) -> Result<EnvSetResponse> {
        self.client
            .post(
                &format!("/sandbox/sandboxes/{}/env", self.sandbox_id),
                Some(&serde_json::json!({ "vars": vars })),
            )
            .await
    }

    pub async fn list(&self) -> Result<EnvListResponse> {
        self.client
            .get(&format!("/sandbox/sandboxes/{}/env", self.sandbox_id))
            .await
    }

    pub async fn delete(&self, key: &str) -> Result<EnvDeleteResponse> {
        self.client
            .post(
                &format!("/sandbox/sandboxes/{}/env/delete", self.sandbox_id),
                Some(&serde_json::json!({ "key": key })),
            )
            .await
    }
}
